dali/test/python/test_operator_numpy_reader.py

```diff
@@ -122,3 +230,65 @@ def check_array(filename, shape, typ, fortran_order=False):
 
     # delete temp files
     delete_numpy_file(filename)
+
+
+def check_array_static_slice(filename, shape, slice_anchor, slice_shape, typ,
+                            io_size=0, fortran_order = False, use_mmap = True):
+    # setup file
+    create_numpy_file(filename, shape, typ, fortran_order)
+    
+    for num_threads in [1, 2, 4, 8]:
+	# load with numpy reader
+        pipe = NumpyReaderPipeline(path = os.path.dirname(filename),
+                                   path_filter = os.path.basename(filename),
+                                   batch_size = 1,
+                                   num_threads = num_threads,
+                                   device_id = 0,
+                                   slice_anchor = slice_anchor,
+                                   slice_shape = slice_shape,
+                                   io_size = io_size,
+                                   use_mmap = use_mmap)
+        pipe.build()
+        pipe_out = pipe.run()
+        arr_rd = np.squeeze(pipe_out[0].as_array(), axis=0)
+
+        # load manually
+        arr_np = np.load(filename)
+        slab = [slice(x[0], x[0]+x[1]) for x in zip(slice_anchor, slice_shape)]
+
+        print(arr_rd, arr_np)
+        
+        # compare
+        assert_array_equal(arr_rd, arr_np[slab])
+
+    # delete temp files
+    delete_numpy_file(filename)
+
+
+def check_array_dynamic_slice(filename, shape, typ, fortran_order = False, use_mmap = True):
```

Comment:
Maybe you can add some sanity check for `use_mmap = False`?

Reply:
the code should work with use_mmap = True. I can extend the test to test both variants